ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c

```diff
@@ -72,12 +72,15 @@ typedef struct
   nnfw_session *session;
   gchar *model_file;
   accl_hw accelerator;
+
+  NNFW_TYPE in_type[NNS_TENSOR_SIZE_LIMIT];   /**< cached input tensor types */
```

Comment:
How about keeping nnfw_tensorinfo for in/out information?  This will include rank and dimension also.

Reply:
Yes, nnfw_tensorinfo can be used. However, do we need that info?
Dimension is directly available from nnfw_pdata->in/out_info, and rank is not used internally by nnstreamer.

Reply:
gst_tensor_info_get_rank() in nnfw_tensor_info_set() (Maybe set-info is not available now)
We can get the exact rank from nnfw_tensorinfo.
or, removing GstTensorsInfo in_info; and GstTensorsInfo out_info;?

Reply:
We cannot extract the exact rank from nnfw_tensorinfo as the info is provided by the user.
The purpose of nnfw_tensor_info_set() is to generate the new nnfw_tensor_info from the provided GstTensorInfo. 
Using the cached nnfw_tensorinfo for this wont be possible.